From 63517c13e2301a0bf25fc3dddabdd695aa02e7c7 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 19 Feb 2024 12:19:31 +0200
Subject: [PATCH] [SW-137122] gaudi3: align bss segment to dword

The issue is bss segment size is not aligned to dword.
Because of this, start up code is skipping initialization
of unaligned address. Here one of the static variable got
corrupted and lead to unknown behaviour.

So we are making sure bss segment size must be aligned
to dword and startup code initialize the whole bss segment
properly.

Change-Id: I6622acb408f3cfdeced6c509f133d02d136938f0
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/pcie_erom.lds.S b/app/ext_rom/pcie_erom.lds.S
index fac5de6..8c43c57 100644
--- a/app/ext_rom/pcie_erom.lds.S
+++ b/app/ext_rom/pcie_erom.lds.S
@@ -22,6 +22,7 @@
 		. = ALIGN(8);
 		__bss_start = . ;
 		*(.bss*)
+		. = ALIGN(4);
 		__bss_end = . ;
 	} > .sram
 
